### PR TITLE
[10][ADD] product exception

### PR DESCRIPTION
**Pull request by hparfr**
Ensure products configuration against rules

Leverage base_exception for products.

### CODE REVIEW

File: product_exception/demo/product_exception.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (nikul-serpentcs) [inline product_exception/demo/product_exception.xml]: @hparfr   `<odoo noupdate="1">`

File: product_exception/models/product.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    rule_group = fields.Selection(
+        selection_add=[('product', 'Product')],
+        default='product',
+    )
+
+    @api.multi
+    def check_exception(self):
+        if self.detect_exceptions():
+            return self._popup_exceptions()
+        else:
```
Comment (nikul-serpentcs) [inline product_exception/models/product.py]: no need `else`

```
if self.detect_exceptions():
            return self._popup_exceptions()
return True
```

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Nitpicking

**Comment by mourad-ehm** [comment]
Why only product.template model is concerned by rules ? (but not product.product)

**Comment by hparfr** [comment]
I didn't need it for my business cases.
Most of the cases the validation can take place and be managed from the template even if you target variants.

By the way, there is the concept of 'lines' in base_exception. Like sale order / sale order line. If you can't do what you need from the template, you may investigate this.

**Review by mourad-ehm — APPROVED** [review]
Code review

**Review by bealdav — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@hparfr Could you squash commits ?

**Comment by hparfr** [comment]
Squashed as requested by @rousseldenis

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by rousseldenis** [comment]
@nikul-serpentcs Could you update your review?

File: product_exception/README.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+See base_exception for more information on rules.
+
+Usage
+=====
+
+A button "Check Exception" is present on the header of the product form.
+When clicked it will check this product against the rules. Results are displayed
+on the "General Information" page.
+
+Known issues / Roadmap
```
Comment (nikul-serpentcs) [inline product_exception/README.rst:53]: can you please remove blank **Known issues / Roadmap**

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM:

**Pull request merged**